trino/client.py

```diff
@@ -1128,14 +1128,26 @@ def map(self, values: List[Any]) -> Optional[List[Any]]:
         return [self.mapper.map(value) for value in values]
 
 
+class NamedRowTuple(tuple):
+    """Custom tuple class as namedtuple doesn't support missing or duplicate names"""
+    def __new__(cls, values, names):
+        return super().__new__(cls, values)
+
+    def __init__(self, values, names):
+        for name, value in zip(names, values):
+            if name is not None and not hasattr(self, name):
+                setattr(self, name, value)
```

Comment:
Could type information be added as well for each row? We often convert the results of our trino queries into polars dataframes, and by providing type information polars can infer the schema so we can stream the data back to arrow IPC. Should be possible by updating `self.__annotations__[name]` with the relevant type. Can open a PR for this functionality as well.

Reply:
Interesting! I'll give that a try.

Reply:
I added names and types annotations, please have a look at the tests.

Reply:
I didn't want to go with `self.__annotations__[name]` as that doesn't support duplicate column names.

Reply:
Hmmm yeah, that's a big problem... based on what I just read in the source of pola-rs/polars#5057, they don't support duplicate keys either since annotations are assumed to be a `dict` per [python guidance on __annotations__](https://docs.python.org/3/howto/annotations.html). We don't write queries with duplicate columns by design, so would likely require some processing on our end.

Reply:
On second thought I prefer to remove the annotations and first merge the attributes support. Can't you extract the information from `cursor.description` property?

Reply:
That's what we've been doing in the past and it's worked perfectly fine, just saw this PR and figured it might be decent to have it implemented directly in Trino :) but it seems like it may be more work than I first believe wrt handled duplicated column names